### PR TITLE
Mat 6038 validate imported test case

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-6038](https://jira.cms.gov/browse/MAT-6038)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java [hunk 1]
```diff
@@ -72,37 +166,130 @@ public void testGetGroupsWithValidPopulationsRemovePopulationsWithoutDefinition(
...
+    assertNotNull(changedGroups.get(0));
+    assertNotNull(changedGroups.get(0).getPopulations());
+    assertThat(changedGroups.get(0).getPopulations().size(), is(5));
+  }
+
+  @Test
+  public void testMatchCriteriaGroupsSuccessAllMatch() {
+    boolean result =
+        testCaseServiceUtil.matchCriteriaGroups(testCaseGroupPopulations, List.of(group), testCase);
+    assertEquals(true, result);
```
Comment (RohitKandimalla) [inline src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java]: not a biggie, but we can do assertTrue

File: src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java [hunk 2]
```diff
@@ -29,6 +36,93 @@ public class TestCaseServiceUtilTest {
...
+          .name(PopulationType.NUMERATOR_EXCLUSION)
+          .expected("5")
+          .build();
+
+  private List<TestCaseGroupPopulation> testCaseGroupPopulations = new ArrayList<>();
+  TestCaseGroupPopulation testCaseGroupPopulation1 = null;
+  TestCaseGroupPopulation testCaseGroupPopulation2 = null;
+  TestCaseGroupPopulation testCaseGroupPopulation3 = null;
+  TestCaseGroupPopulation testCaseGroupPopulation4 = null;
+  TestCaseGroupPopulation testCaseGroupPopulation5 = null;
```
Comment (RohitKandimalla) [inline src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java]: Looks like testCaseGroupPopulation3, 4 & 5 are never used.

File: src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java [hunk 3]
```diff
@@ -488,8 +499,107 @@ public void testGetTestCaseGroupPopulationsFromMeasureReportNoCount()
...
..."display\":\"Patrick Pump\"},\"subject\":{\"reference\":\"Patient/0e3be52f-723e-4df4-a584-337daa19e259\"}}},{\"fullUrl\":\"Coverage/1\",\"resource\":{\"resourceType\":\"Coverage\",\"beneficiary\":{\"reference\":\"Patient/0e3be52f-723e-4df4-a584-337daa19e259\"},\"id\":\"1\",\"meta\":{\"profile\":[\"http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-coverage\"]},\"payor\":[{\"reference\":\"Organization/123456\"}],\"status\":\"active\"}},{\"fullUrl\":\"Organization/123456\",\"resource\":{\"resourceType\":\"Organization\",\"active\":true,\"address\":[{\"use\":\"billing\",\"type\":\"postal\",\"line\":[\"P.O. Box 660044\"],\"city\":\"Dallas\",\"state\":\"TX\",\"postalCode\":\"75266-0044\",\"country\":\"USA\"}],\"id\":\"123456\",\"identifier\":[{\"use\":\"temp\",\"system\":\"urn:oid:2.16.840.1.113883.4.4\",\"value\":\"21-3259825\"}],\"meta\":{\"profile\":[\"http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-organization\"]},\"name\":\"Blue Cross Blue Shield of Texas\",\"telecom\":[{\"system\":\"phone\",\"value\":\"(+1) 972-766-6900\"}],\"type\":[{\"coding\":[{\"system\":\"http://terminology.hl7.org/CodeSystem/organization-type\",\"code\":\"pay\",\"display\":\"Payer\"}]}]}}]}";
```
Comment (RohitKandimalla) [inline src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java]: Do you think we can add these multiple test JSON strings as a resource and then load them ?

File: src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java [hunk 4]
```diff
@@ -488,8 +499,107 @@ public void testGetTestCaseGroupPopulationsFromMeasureReportNoCount()
...
+    TestCase tc1 =
+        TestCase.builder().id("TC1").name("TC1").patientId(UUID.randomUUID()).json(json).build();
+    // String baseUrl = "https://myorg.com";
+    // ReflectionTestUtils.setField(testCaseService, "madieJsonResourcesBaseUri", baseUrl);
```
Comment (RohitKandimalla) [inline src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java]: We could remove these ?

File: src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java]: @RohitKandimalla Done.

File: src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java]: @RohitKandimalla Removed.

File: src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java (hunk 3 above)
Comment (sb-cecilialiu) [reply src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java]: @RohitKandimalla Done as well, added the resource test json files.

File: src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java (hunk 4 above)
Comment (sb-cecilialiu) [reply src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java]: @RohitKandimalla Removed. Thanks so much Rohit!

File: src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java (hunk 1 above)
Comment (RohitKandimalla) [reply src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java]: It looks like we have quite a few.

File: src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java (hunk 3 above)
Comment (RohitKandimalla) [reply src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java]: I could still see this

File: src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/test/java/cms/gov/madie/measure/utils/TestCaseServiceUtilTest.java]: @RohitKandimalla replaced all of them.

File: src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java (hunk 3 above)
Comment (sb-cecilialiu) [reply src/test/java/cms/gov/madie/measure/utils/QiCoreJsonUtilTest.java]: @RohitKandimalla Added another resource file for this. Thanks Rohit!

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**